Lib/test/test_descr.py

```diff
@@ -5003,6 +5003,19 @@ class Child(Parent):
         gc.collect()
         self.assertEqual(Parent.__subclasses__(), [])
 
+    def test_attr_raise_through_property(self):
+        # add test case for gh-103272
+        class A:
+            def __getattr__(self, name):
+                raise ValueError
```

Comment:
One last nit: I think we can use an error message here and check it in `assertRaisesRegex`. Right now, technically some different `ValueError` might fit :(